packages/eas-cli/src/submissions/ios/AppProduce.ts

```diff
@@ -62,9 +64,86 @@ Learn more here: https://expo.fyi/bundle-identifier`
     language: sanitizeLanguage(language) ?? 'English',
   };
 
+  if (USE_APPLE_UTILS) {
+    return await runProduceExperimentalAsync(options);
+  }
+
   return await runProduceAsync(options);
 }
 
+async function isProvisioningAvailableAsync(requestCtx: RequestContext): Promise<boolean> {
+  const session = await Session.getAnySessionInfo();
+  const username = session?.user.emailAddress;
+  const [user] = await User.getAsync(requestCtx, { query: { filter: { username } } });
+  return user.attributes.provisioningAllowed;
+}
+
+async function runProduceExperimentalAsync(options: ProduceOptions): Promise<AppStoreResult> {
+  const {
+    appleId,
+    appleTeamId,
+    bundleIdentifier: bundleId,
+    appName,
+    language,
+    companyName,
+  } = options;
+
+  const authCtx = await authenticateAsync({
+    appleId,
+    team: { id: appleTeamId },
+  } as any);
+  const requestCtx = getRequestContext(authCtx);
+
+  log.addNewLineIfNone();
+
+  if (await isProvisioningAvailableAsync(requestCtx)) {
+    log(`Ensuring that bundle ID "${bundleId}" is registered on Apple Dev Center...`);
+    await App.ensureBundleIdExistsAsync(requestCtx, {
+      name: appName,
+      bundleId,
+    });
+  } else {
+    log.warn(
+      `Provisioning is not available for user "${authCtx.appleId}", skipping bundle identifier check.`
+    );
+  }
+
+  log(`Checking App Store Connect for "${appName}" (${bundleId})...`);
+  let app = await App.findAsync(requestCtx, { bundleId });
+
+  if (!app) {
+    log(`Creating app "${appName}" (${bundleId}) on App Store Connect...`);
+    try {
+      app = await App.createAsync(requestCtx, {
+        bundleId,
+        name: appName,
+        // TODO: Convert values like "English" to "en-US"
+        primaryLocale: language,
+        companyName,
+      });
+    } catch (error) {
+      if (error.message.match(/An App ID with Identifier '(.*)' is not available/)) {
+        log.warn(
+          `\nThe bundle identifier "${bundleId}" is not available, please change it in your app config and try again.\n`
```

Comment:
Should specify that the bundle id is not available to `"provider.name" (provider.providerId)` in case maybe the user chose the wrong provider.